client/src/components/screens/Patient/components/FilesTab/index.tsx

```diff
@@ -1,233 +1,66 @@
-      index: string;
-    },
-    metaData: PatientResponse,
-  ) => {
-    const option = [];
-
-    option.push(
-      (
-        <Menu.Item>
-          <Button
-            className="link--underline"
-            onClick={() => dispatch(getPatientFileURL(url.file))}
-            target="_blank"
-            type="link"
-          >
-            { intl.get('screen.patient.details.file.download.file') }
-          </Button>
-        </Menu.Item>
-      ),
-      (
-        <Menu.Item>
-          <Button
-            className="link--underline"
-            onClick={() => {
-              setDocumentReference(JSON.stringify(metaData));
-              setIsOpenModal(true);
-            }}
-            type="link"
-          >
-            Metadata
-          </Button>
-        </Menu.Item>
-      ),
-    );
-    if (format === 'CRAM' || format === 'VCF') {
-      option.push(
-        (
-          <Menu.Item>
-            <Button
-              className="link--underline"
-              onClick={() => dispatch(getPatientFileURL(url.index))}
-              target="_blank"
-              type="link"
-            >
-
-              Index
-            </Button>
-          </Menu.Item>
-        ),
-      );
-    }
-
-    const menu = (
-      <Menu>
-        {
-          option
-        }
-      </Menu>
-    );
-
-    return (
-      <div>
-        <Dropdown
-          className="files-tab__dropdownAction"
-          overlay={menu}
-        >
-          <Button type="link">
-            { intl.get('screen.patient.details.file.download') } <DownOutlined />
-          </Button>
-        </Dropdown>
-
-      </div>
-    );
-  };
-
-  const getData = () => {
-    filesResults.results.docs.forEach((element: PatientResponse) => {
-      const {
-        size, title,
-      } = element.content[0].attachment;
-      const { type } = element;
-      const { format } = element.content[0];
-      const sizeWithUnity = formatBytes(Number(size));
-      const specimen = element.aliquot.resource[0].sample[0].resource.external_id;
-      const aliquot = element.aliquot.resource[0].external_id;
-      const prescription = element.task.focus.reference.split('/')[1];
-      const date = element.task.runDate;
-
-      const url = {
-        file: element.content[0].attachment.url,
-        index: element.content.length > 1 ? element.content[1].attachment.url : '',
-      };
-
-      const dataLine = {
-        action: getDropdownOption(format, url, element),
-        aliquot,
-        date,
-        format,
-        prescription,
-        sample: specimen,
-        size: sizeWithUnity,
-        title,
-        type,
-      };
-
-      dataSource.push(dataLine);
-    });
-  };
-
-  if (!filesResults.results.docs) {
+  if (!prescriptionFileResults.loading &&!prescriptionFileResults.results.docs) {
     return <EmptyCard />
   }
-  getData();
+  const dataSource =!prescriptionFileResults.loading ? makeRowsFromFilesResults(prescriptionFileResults) : [];
```

Comment:
no need to do that. put directly makeRowsFromFilesResults(prescriptionFileResults) as a prop in <Table/>

Reply:
this should be set by default in actions.

le type dans https://github.com/Ferlab-Ste-Justine/clin-frontend/blob/c06065f38fdadab3911867a2be25c99e0143e58f/client/src/helpers/search/types.ts#L96 n'est même pas optionnel. Alors il doit retourner au minimum un array vide.

makeRowsFromFilesResults peut donc rouler sans problème. En gros, ta condition ne change rien au résultat possible